File: Payhub - SDK/java_SDK/PayHubSDK/src/main/java/com/payhub/ws/api/TransactionReportInformation.java
/**
 * 
 */
package com.payhub.ws.api;

import java.util.List;

import com.fasterxml.jackson.annotation.JsonInclude;

/**
 * @author agustin
 *
 */
@JsonInclude(JsonInclude.Include.NON_NULL)  
public class TransactionReportInformation {
	private String transactionID;
	private String batchID;
	private String transactionDate;
	private String customerName;
	private String cardType;
	private String cardLast4Digits;
	private String cardToken;
	private String responseCode;
	private String responseText;
	private String transactionType;
	private String amount;
	private String authAmount;
	private String swiped;
	private String source;
	private String phoneNumber;
	private String email;
	private String note;
	private String transactionStatus;
	private List<Errors> errors;
	/**
	 * @return the transactionID
	 */
	public String getTransactionID() {
		return transactionID;
	}
	/**
	 * @param transactionID the transactionID to set
	 */
	public void setTransactionID(String transactionID) {
		this.transactionID = transactionID;
	}
	/**
	 * @return the batchID
	 */
	public String getBatchID() {
		return batchID;
	}
	/**
	 * @param batchID the batchID to set
	 */
	public void setBatchID(String batchID) {
		this.batchID = batchID;
	}
	/**
	 * @return the transactionDate
	 */
	public String getTransactionDate() {
		return transactionDate;
	}
	/**
	 * @param transactionDate the transactionDate to set
	 */
	public void setTransactionDate(String transactionDate) {
		this.transactionDate = transactionDate;
	}
	/**
	 * @return the customerName
	 */
	public String getCustomerName() {
		return customerName;
	}
	/**
	 * @param customerName the customerName to set
	 */
	public void setCustomerName(String customerName) {
		this.customerName = customerName;
	}
	/**
	 * @return the cardType
	 */
	public String getCardType() {
		return cardType;
	}
	/**
	 * @param cardType the cardType to set
	 */
	public void setCardType(String cardType) {
		this.cardType = cardType;
	}
	/**
	 * @return the cardLast4Digits
	 */
	public String getCardLast4Digits() {
		return cardLast4Digits;
	}
	/**
	 * @param cardLast4Digits the cardLast4Digits to set
	 */
	public void setCardLast4Digits(String cardLast4Digits) {
		this.cardLast4Digits = cardLast4Digits;
	}
	/**
	 * @return the cardToken
	 */
	public String getCardToken() {
		return cardToken;
	}
	/**
	 * @param cardToken the cardToken to set
	 */
	public void setCardToken(String cardToken) {
		this.cardToken = cardToken;
	}
	/**
	 * @return the responseCode
	 */
	public String getResponseCode() {
		return responseCode;
	}
	/**
	 * @param responseCode the responseCode to set
	 */
	public void setResponseCode(String responseCode) {
		this.responseCode = responseCode;
	}
	/**
	 * @return the responseText
	 */
	public String getResponseText() {
		return responseText;
	}
	/**
	 * @param responseText the responseText to set
	 */
	public void setResponseText(String responseText) {
		this.responseText = responseText;
	}
	/**
	 * @return the transactionType
	 */
	public String getTransactionType() {
		return transactionType;
	}
	/**
	 * @param transactionType the transactionType to set
	 */
	public void setTransactionType(String transactionType) {
		this.transactionType = transactionType;
	}
	/**
	 * @return the amount
	 */
	public String getAmount() {
		return amount;
	}
	/**
	 * @param amount the amount to set
	 */
	public void setAmount(String amount) {
		this.amount = amount;
	}
	/**
	 * @return the authAmount
	 */
	public String getAuthAmount() {
		return authAmount;
	}
	/**
	 * @param authAmount the authAmount to set
	 */
	public void setAuthAmount(String authAmount) {
		this.authAmount = authAmount;
	}
	/**
	 * @return the swiped
	 */
	public String getSwiped() {
		return swiped;
	}
	/**
	 * @param swiped the swiped to set
	 */
	public void setSwiped(String swiped) {
		this.swiped = swiped;
	}
	/**
	 * @return the source
	 */
	public String getSource() {
		return source;
	}
	/**
	 * @param source the source to set
	 */
	public void setSource(String source) {
		this.source = source;
	}
	/**
	 * @return the phoneNumber
	 */
	public String getPhoneNumber() {
		return phoneNumber;
	}
	/**
	 * @param phoneNumber the phoneNumber to set
	 */
	public void setPhoneNumber(String phoneNumber) {
		this.phoneNumber = phoneNumber;
	}
	/**
	 * @return the email
	 */
	public String getEmail() {
		return email;
	}
	/**
	 * @param email the email to set
	 */
	public void setEmail(String email) {
		this.email = email;
	}
	/**
	 * @return the note
	 */
	public String getNote() {
		return note;
	}
	/**
	 * @param note the note to set
	 */
	public void setNote(String note) {
		this.note = note;
	}
	/**
	 * @return the transactionStatus
	 */
	public String getTransactionStatus() {
		return transactionStatus;
	}
	/**
	 * @param transactionStatus the transactionStatus to set
	 */
	public void setTransactionStatus(String transactionStatus) {
		this.transactionStatus = transactionStatus;
	}
	/**
	 * @return the errors
	 */
	public List<Errors> getErrors() {
		return errors;
	}
	/**
	 * @param errors the errors to set
	 */
	public void setErrors(List<Errors> errors) {
		this.errors = errors;
	}
	
	
	
}

File: Payhub - SDK/php_SDK/com/payhub/ws/api/TransactionReportInformation.php
<?php

/**
 * Created by PhpStorm.
 * User: agustin
 * Date: 24/07/2015
 * Time: 11:08
 */
class TransactionReportInformation
{
    public $transactionID;
    public $batchID;
    public $transactionDate;
    public $customerName;
    public $cardType;
    public $cardLast4Digits;
    public $cardToken;
    public $responseCode;
    public $responseText;
    public $transactionType;
    public $amount;
    public $authAmount;
    public $swiped;
    public $source;
    public $phoneNumber;
    public $email;
    public $note;
    public $transactionStatus;
    public $errors;
	public $voidedBy;
	public $refundedBy;
	public $isCaptured;
    public $customerId;
    public $recurringBillId;

    /**
     * @return mixed
     */
    public function getTransactionID()
    {
        return $this->transactionID;
    }

    /**
     * @param mixed $transactionID
     */
    public function setTransactionID($transactionID)
    {
        $this->transactionID = $transactionID;
    }

    /**
     * @return mixed
     */
    public function getBatchID()
    {
        return $this->batchID;
    }

    /**
     * @param mixed $batchID
     */
    public function setBatchID($batchID)
    {
        $this->batchID = $batchID;
    }

    /**
     * @return mixed
     */
    public function getTransactionDate()
    {
        return $this->transactionDate;
    }

    /**
     * @param mixed $transactionDate
     */
    public function setTransactionDate($transactionDate)
    {
        $this->transactionDate = $transactionDate;
    }

    /**
     * @return mixed
     */
    public function getCustomerName()
    {
        return $this->customerName;
    }

    /**
     * @param mixed $customerName
     */
    public function setCustomerName($customerName)
    {
        $this->customerName = $customerName;
    }

    /**
     * @return mixed
     */
    public function getCardType()
    {
        return $this->cardType;
    }

    /**
     * @param mixed $cardType
     */
    public function setCardType($cardType)
    {
        $this->cardType = $cardType;
    }

    /**
     * @return mixed
     */
    public function getCardLast4Digits()
    {
        return $this->cardLast4Digits;
    }

    /**
     * @param mixed $cardLast4Digits
     */
    public function setCardLast4Digits($cardLast4Digits)
    {
        $this->cardLast4Digits = $cardLast4Digits;
    }

    /**
     * @return mixed
     */
    public function getCardToken()
    {
        return $this->cardToken;
    }

    /**
     * @param mixed $cardToken
     */
    public function setCardToken($cardToken)
    {
        $this->cardToken = $cardToken;
    }

    /**
     * @return mixed
     */
    public function getResponseCode()
    {
        return $this->responseCode;
    }

    /**
     * @param mixed $responseCode
     */
    public function setResponseCode($responseCode)
    {
        $this->responseCode = $responseCode;
    }

    /**
     * @return mixed
     */
    public function getResponseText()
    {
        return $this->responseText;
    }

    /**
     * @param mixed $responseText
     */
    public function setResponseText($responseText)
    {
        $this->responseText = $responseText;
    }

    /**
     * @return mixed
     */
    public function getTransactionType()
    {
        return $this->transactionType;
    }

    /**
     * @param mixed $transactionType
     */
    public function setTransactionType($transactionType)
    {
        $this->transactionType = $transactionType;
    }

    /**
     * @return mixed
     */
    public function getAmount()
    {
        return $this->amount;
    }

    /**
     * @param mixed $amount
     */
    public function setAmount($amount)
    {
        $this->amount = $amount;
    }

    /**
     * @return mixed
     */
    public function getAuthAmount()
    {
        return $this->authAmount;
    }

    /**
     * @param mixed $authAmount
     */
    public function setAuthAmount($authAmount)
    {
        $this->authAmount = $authAmount;
    }

    /**
     * @return mixed
     */
    public function getSwiped()
    {
        return $this->swiped;
    }

    /**
     * @param mixed $swiped
     */
    public function setSwiped($swiped)
    {
        $this->swiped = $swiped;
    }

    /**
     * @return mixed
     */
    public function getSource()
    {
        return $this->source;
    }

    /**
     * @param mixed $source
     */
    public function setSource($source)
    {
        $this->source = $source;
    }

    /**
     * @return mixed
     */
    public function getPhoneNumber()
    {
        return $this->phoneNumber;
    }

    /**
     * @param mixed $phoneNumber
     */
    public function setPhoneNumber($phoneNumber)
    {
        $this->phoneNumber = $phoneNumber;
    }

    /**
     * @return mixed
     */
    public function getEmail()
    {
        return $this->email;
    }

    /**
     * @param mixed $email
     */
    public function setEmail($email)
    {
        $this->email = $email;
    }

    /**
     * @return mixed
     */
    public function getNote()
    {
        return $this->note;
    }

    /**
     * @param mixed $note
     */
    public function setNote($note)
    {
        $this->note = $note;
    }

    /**
     * @return mixed
     */
    public function getTransactionStatus()
    {
        return $this->transactionStatus;
    }

    /**
     * @param mixed $transactionStatus
     */
    public function setTransactionStatus($transactionStatus)
    {
        $this->transactionStatus = $transactionStatus;
    }

    /**
     * @return mixed
     */
    public function getErrors()
    {
        return $this->errors;
    }

    /**
     * @param mixed $errors
     */
    public function setErrors($errors)
    {
        $this->errors = $errors;
    }

    /**
     * @return mixed
     */
    public function getVoidedBy()
    {
        return $this->voidedBy;
    }

    /**
     * @param mixed $voidedBy
     */
    public function setVoidedBy($voidedBy)
    {
        $this->voidedBy = $voidedBy;
    }

    /**
     * @return mixed
     */
    public function getRefundedBy()
    {
        return $this->refundedBy;
    }

    /**
     * @param mixed $refundedBy
     */
    public function setRefundedBy($refundedBy)
    {
        $this->refundedBy = $refundedBy;
    }

    /**
     * @return mixed
     */
    public function getIsCaptured()
    {
        return $this->isCaptured;
    }

    /**
     * @param mixed $isCaptured
     */
    public function setIsCaptured($isCaptured)
    {
        $this->isCaptured = $isCaptured;
    }

    /**
     * @return mixed
     */
    public function getCustomerId()
    {
        return $this->customerId;
    }

    /**
     * @param mixed $customerId
     */
    public function setCustomerId($customerId)
    {
        $this->customerId = $customerId;
    }

    /**
     * @return mixed
     */
    public function getRecurringBillId()
    {
        return $this->recurringBillId;
    }

    /**
     * @param mixed $recurringBillId
     */
    public function setRecurringBillId($recurringBillId)
    {
        $this->recurringBillId = $recurringBillId;
    }

    

    public static function fromArray($data){
        if(!is_null($data)) {
            $err=null;
            $tri = new TransactionReportInformation();
            foreach ($data as $key => $value) {
                if (property_exists(get_class($tri), $key)) {
                   $tri->{$key} = $value;
                }else{
                    $err[]=$value;
                }
            }
            if(!is_null($err)){
                $tri->errors=Errors::fromArray($err);
            }
            return $tri;
        }
    }
}